pkg/diff/sql_generator.go

```diff
@@ -432,7 +465,27 @@ func (schemaSQLGenerator) Alter(diff schemaDiff) ([]Statement, error) {
 		return nil, fmt.Errorf("unioning table and trigger graphs: %w", err)
 	}
 
-	return tableGraphs.toOrderedStatements()
+	statements, err := tableGraphs.toOrderedStatements()
+	if err != nil {
+		return nil, fmt.Errorf("getting ordered statements from tableGraph: %w", err)
+	}
+
+	return mergeStatementsInOrder(
+		extensionStatements.Adds,
+		extensionStatements.Alters,
+		statements, // these contain every other statement
+		extensionStatements.Deletes,
+	), nil
+}
+
+// mergeStatementsInOrder returns a merged list of the list of statements in the order
+// they're provided.
+func mergeStatementsInOrder(listOfStatements ...[]Statement) []Statement {
```

Comment:
I'm leaning towards getting rid of this and doing an old fashioned set of appends, [like here](https://github.com/stripe/pg-schema-diff/blob/ff76172f/pkg/diff/sql_generator.go#L558-L565) . Reduces the indirection a bit